src/viperleed/gui/measure/classes/downloader.py

```diff
@@ -0,0 +1,189 @@
+        # case with parent-child relationships) seems to be broken in
+        # Qt5. We move self._network explicitly in self.moveToThread.
+        # Assigning a parent here would break the overridden
+        # self.moveToThread, as children with a parent cannot be moved
+        # 'independently'.
+        self._network = qtn.QNetworkAccessManager()
+        self._network.setTransferTimeout(timeout=2000)
+        self._thread = None
+        self._worker = None
+
+    def download(self, url):
+        """Start a download from url.
+
+        Choose QtNetwork or requests dynamically depending on system
+        availability. QtNetwork is preferred. Missing QtNetwork support
+        is detected automatically.
+
+        Parameters
+        ----------
+        url : str
+            Link to data to download.
+
+        Returns
+        -------
+        None.
+        """
+        if not qtn.QSslSocket.supportsSsl():
+            # Cannot use QtNetwork, use requests module instead.
+            self._download_with_requests(url)
+            return
+
+        request = qtn.QNetworkRequest(qtc.QUrl(url))
+        # The RedirectAttribute needs to be set in order to get the
+        # Arduino CLI file via http because github redirects from https
+        # to http .
+        request.setAttribute(qtn.QNetworkRequest.FollowRedirectsAttribute,
+                             True)
+        self._network.finished.connect(self._on_qt_finished)
+        self._network.get(request)
+
+    def moveToThread(self, thread):     # pylint: disable=invalid-name
+        """Move self and self._network to thread."""
+        super().moveToThread(thread)
+        self._network.moveToThread(thread)
+
+    @qtc.pyqtSlot(bytes)
+    @qtc.pyqtSlot()
+    def _cleanup_thread(self, *_):
+        """Delete thread and worker objects."""
+        if self._thread:
+            self._thread.quit()
+            self._thread.wait()
+            self._thread.deleteLater()
+
+        if self._worker:
+            self._worker.deleteLater()
+
+        self._worker = None
+        self._thread = None
+
+    def _download_with_requests(self, url):
+        """Download data with requests module.
+
+        Parameters
+        ----------
+        url : str
+            Link to data to download.
+
+        Emits
+        -----
+        requests_module_not_found
+            If requests module is not installed.
+        """
+        try:
+            importlib.import_module('requests')
+        except ImportError:
+            self.requests_module_not_found.emit()
+            return
+        self._thread = qtc.QThread(self)
+        self._worker = _RequestsWorker(url)
+        self._worker.moveToThread(self._thread)
+
+        # Wire signals
+        self._worker.finished.connect(self._cleanup_thread)
+        self._worker.failed.connect(self._cleanup_thread)
+        self._worker.finished.connect(self.download_finished.emit)
+        self._worker.failed.connect(self.download_failed.emit)
```

Comment:
```suggestion
        self._worker.finished.connect(self.download_finished)
        self._worker.failed.connect(self.download_failed)
```

I think that signals can be connected to signals directly.